File: Data Structures & Algorithms/longest-consecutive-sequence/submission-0.cpp
class Solution {
public:
    int longestConsecutive(vector<int>& nums) {
        unordered_set<int> s(nums.begin(), nums.end());
        int longest = 0;
        for (int i : s) {
            if (!s.contains(i - 1)) {
                int num = i + 1;
                int tmp = 1;
                while (s.contains(num)) {
                    tmp++;
                    num++;
                }
                longest = max(longest, tmp);
            }
        }
        return longest;
    }
};

File: Data Structures & Algorithms/house-robber/submission-0.py
class Solution:
    def robR(self,houseI : int):
        if (houseI >= len(self.nums)):
            return 0
        if houseI in self.prevInfo:
            return self.prevInfo[houseI]
        self.prevInfo[houseI] = max(self.robR(houseI+1),self.robR(houseI+2) + self.nums[houseI])
        return self.prevInfo[houseI]
    def rob(self, nums: List[int]) -> int:
        self.prevInfo = {}
        self.nums = nums
        return self.robR(0)

    
